src/api/app/controllers/attribute_controller.rb

```diff
@@ -34,134 +36,76 @@ def index
     render xml: xml
   end
 
-  # /attribute/:namespace/_meta
-  def namespace_definition
-    if params[:namespace].nil?
-      raise MissingParameterError, "parameter 'namespace' is missing"
+  def show_namespace_definition
+    if (@an = AttribNamespace.where(name: ensure_namespace).select(:id, :name).first)
```

Comment:
This sounds like a `require_namespace` `before_filter` you repeat this quite often below inside `ensure_namespace`. Please decouple this.